pkg/controller/installation/products/ups/reconciler_test.go

```diff
@@ -0,0 +1,221 @@
+		},
+		WriteConfigFunc: func(config config.ConfigReadable) error {
+			return nil
+		},
+	}
+}
+
+func errorConfigMock() *config.ConfigReadWriterMock {
+	return &config.ConfigReadWriterMock{
+		ReadUpsFunc: func() (ups *config.Ups, e error) {
+			return config.NewUps(config.ProductConfig{}), nil
+		},
+		WriteConfigFunc: func(config config.ConfigReadable) error {
+			return errors.New("some error")
+		},
+	}
+}
+
+func basicRouteMock() *v1Route.Route {
+	return &v1Route.Route{
+		TypeMeta: v1.TypeMeta{
+			Kind:       "Route",
+			APIVersion: "v1",
+		},
+		ObjectMeta: v1.ObjectMeta{
+			Namespace: "ups",
+			Name:      defaultRoutename,
+		},
+	}
+}
+
+func TestReconciler_ReconcileCustomResource(t *testing.T) {
+	scheme := runtime.NewScheme()
+	upsv1alpha1.SchemeBuilder.AddToScheme(scheme)
+
+	cases := []struct {
+		Name           string
+		client         client.Client
+		FakeConfig     *config.ConfigReadWriterMock
+		Installation   *v1alpha1.Installation
+		ExpectErr      bool
+		ExpectedStatus v1alpha1.StatusPhase
+		FakeMPM        *marketplace.MarketplaceInterfaceMock
+	}{
+		{
+			Name:           "UPS Test: test custom resource is reconciled and phase complete returned",
+			ExpectedStatus: v1alpha1.PhaseCompleted,
+			FakeMPM:        &marketplace.MarketplaceInterfaceMock{},
+			Installation:   &v1alpha1.Installation{},
+			FakeConfig:     basicConfigMock(),
+			client: fake.NewFakeClientWithScheme(scheme,
+				&upsv1alpha1.UnifiedPushServer{
+					ObjectMeta: v1.ObjectMeta{
+						Namespace: "ups",
+						Name:      "ups",
+					},
+					Status: upsv1alpha1.UnifiedPushServerStatus{
+						Phase: upsv1alpha1.PhaseComplete,
+					},
+				}),
+		},
+		{
+			Name:           "UPS Test: Phase failed when error in creating custom resource",
+			ExpectedStatus: v1alpha1.PhaseFailed,
+			FakeMPM:        &marketplace.MarketplaceInterfaceMock{},
+			Installation:   &v1alpha1.Installation{},
+			FakeConfig:     basicConfigMock(),
+			client: &moqclient.SigsClientInterfaceMock{
+				GetFunc: func(ctx context.Context, key types.NamespacedName, obj runtime.Object) error {
+					return k8serr.NewNotFound(schema.GroupResource{Group: upsv1alpha1.SchemeGroupVersion.Group, Resource: "UnifiedPushServer"}, "Not found")
+				},
+				CreateFunc: func(ctx context.Context, obj runtime.Object) error {
+					return errors.New("dummy create error")
+				},
+			},
+			ExpectErr: true,
+		},
+		{
+			Name:           "UPS Test: Phase failed when general error in finding custom resource",
+			ExpectedStatus: v1alpha1.PhaseFailed,
+			FakeMPM:        &marketplace.MarketplaceInterfaceMock{},
+			Installation:   &v1alpha1.Installation{},
+			FakeConfig:     basicConfigMock(),
+			client: &moqclient.SigsClientInterfaceMock{
+				GetFunc: func(ctx context.Context, key types.NamespacedName, obj runtime.Object) error {
+					return errors.New("General error")
+				},
+				CreateFunc: func(ctx context.Context, obj runtime.Object) error {
```

Comment:
I don't think you need to mock this create function since the parent function errors out before this is called